flink-connectors/flink-connector-aws-kinesis-data-streams/src/test/java/org/apache/flink/connectors/kinesis/testutils/KinesaliteContainer.java

```diff
@@ -99,25 +99,41 @@ public Properties getHostProperties() {
     }
 
     /** Returns the client to access the container from outside the docker network. */
-    public KinesisAsyncClient getContainerClient() throws URISyntaxException {
-        return getClient(getContainerEndpointUrl());
+    public KinesisAsyncClient getContainerClient(SdkAsyncHttpClient httpClient)
```

Comment:
I'd suggest renaming the methods to start with a `createXXX` prefix, so it's explicit that these need to be closed and EVERY SINGLE CALL to them creates a new resource.

Reply:
If you are worried about backwards compatibility, we do not guarantee this for test code. Suggesting renaming it to force/suggest external consumers to upgrade and close the client too, if not already.

Reply:
done.